src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -2036,8 +2037,12 @@ def list_resources(cmd, resource_group_name=None,
     return list(resources)
 
 
-def register_provider(cmd, resource_provider_namespace, mg=None, wait=False, accept_terms=None):
-    _update_provider(cmd.cli_ctx, resource_provider_namespace, registering=True, wait=wait, mg_id=mg, accept_terms=accept_terms)
+def register_provider(cmd, resource_provider_namespace, consent_to_permissions=False, mg=None, wait=False, accept_terms=None):
+    properties = None
+    if consent_to_permissions:
+        from azure.mgmt.resource.resources.v2021_04_01.models import ProviderRegistrationRequest, ProviderConsentDefinition
```

Comment:
It is not recommended to directly import the model from package under the specific version. If you plan to bump api-version, then you can use `cmd.get_models` to get those models